be/src/storage/lake/lake_persistent_index.cpp

```diff
@@ -175,8 +180,145 @@ Status LakePersistentIndex::try_replace(size_t n, const Slice* keys, const Index
     return Status::OK();
 }
 
-Status LakePersistentIndex::major_compact(int64_t min_retain_version) {
+std::unique_ptr<sstable::Iterator> LakePersistentIndex::prepare_merging_iterator() {
+    sstable::ReadOptions read_options;
+    std::vector<sstable::Iterator*> iters;
+    auto max_compaction_versions = config::lake_pk_index_sst_max_compaction_versions;
+    iters.reserve(max_compaction_versions);
+    for (auto it = _sstables.begin();
+         it != _sstables.end() && it != _sstables.begin() + config::lake_pk_index_sst_max_compaction_versions; ++it) {
+        sstable::Iterator* iter = (*it)->new_iterator(read_options);
+        iters.emplace_back(iter);
+    }
+    sstable::Options options;
+    sstable::Iterator* iter = sstable::NewMergingIterator(options.comparator, &iters[0], iters.size());
+    std::unique_ptr<sstable::Iterator> iter_ptr = nullptr;
+    iter_ptr.reset(iter);
+    iter_ptr->SeekToFirst();
+    return iter_ptr;
+}
+
+void LakePersistentIndex::build_index_value_vers(const std::string& key,
+                                                 const std::list<IndexValueWithVer>& index_value_vers,
+                                                 sstable::TableBuilder* builder) {
+    if (index_value_vers.empty()) {
+        return;
+    }
+
+    IndexValueWithVerPB index_value_pb;
+    for (const auto& index_value_with_ver : index_value_vers) {
+        index_value_pb.add_versions(index_value_with_ver.first);
+        index_value_pb.add_values(index_value_with_ver.second.get_value());
+    }
+    builder->Add(Slice(key), Slice(index_value_pb.SerializeAsString()));
+}
+
+Status LakePersistentIndex::merge_sstables(std::unique_ptr<sstable::Iterator> iter_ptr,
+                                           sstable::TableBuilder* builder) {
+    auto key = iter_ptr->key().to_string();
+    std::list<IndexValueWithVer> index_value_vers;
+    while (iter_ptr->Valid()) {
+        auto current_key = iter_ptr->key().to_string();
```

Comment:
Swith name between `key` and `current_key`

Reply:
class KeyValMerger {
    void merge(std::string key, std::string value) {
         ... // key vs _key
         build_index_value_vers();
   }
private:
   std::string _key;
}